### PR TITLE
Optimize code structure without affecting functionality

### PR DESCRIPTION
**Pull request by chengxiangdong**
I designed a new development architecture for developing EVS Provisioner, which will also be used to rebuild SFS, SFS-turbo CSI Provisioner in the future, the old code is currently unchanged.

The changes are as follows:
- Extract public functions and some practical utils functions.
- Use a common config and authentication functions.
- Change the way of endpoint generation, no longer depends on API services.

The code has been tested.

### CODE REVIEW

**Review by xiaomoli — COMMENTED** [review]
why not create a `ecs` folder

**Comment by chengxiangdong** [comment]
> why not create a `ecs` folder

why do we need the `ecs` folder? It is not a storage service.

**Comment by xiaomoli** [comment]
Maybe I misunderstood, `func (c *CloudCredentials) EcsV1Client() (*golangsdk.ServiceClient, error) {
	return newServiceClient(c, "ecs", c.Global.Region)
}` in the code, so I think we maybe should  create a folder. such as ` ecs`,`evs`, etc.

File: pkg/config/cloud.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	}
+
+	client, err := openstack.NewClient(ao.IdentityEndpoint)
+	if err != nil {
+		return err
+	}
+
+	// if OS_DEBUG is set, log the requests and responses
+	var osDebug bool
+	if os.Getenv("OS_DEBUG") != "" {
```
Comment (niuzhenguo) [inline pkg/config/cloud.go:114]: is that possible to export this ENV in the csi pod? maybe add a parameters to the cmd to allow users choose whether to enable the debug log when deploy it.

File: pkg/config/cloud.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		return err
+	}
+
+	c.CloudClient = client
+	c.CloudClient.UserAgent.Prepend(UserAgent)
+	return nil
+}
+
+func (c *CloudCredentials) SFSV2Client() (*golangsdk.ServiceClient, error) {
+	return openstack.NewSharedFileSystemV2(c.CloudClient, golangsdk.EndpointOpts{
```
Comment (niuzhenguo) [inline pkg/config/cloud.go:137]: should we keep this or just change to use newServiceClient?

File: pkg/config/cloud.go (hunk 1 above)
Comment (chengxiangdong) [reply pkg/config/cloud.go:114]: add parameter `http-log` to enable the http request debug log

File: pkg/config/cloud.go (hunk 2 above)
Comment (chengxiangdong) [reply pkg/config/cloud.go:137]: the newServiceClient will be used when sfs csi is refactored

File: pkg/config/cloud.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+// CloudCredentials define
+type CloudCredentials struct {
+	Global struct {
+		Cloud     string `gcfg:"cloud"`
+		AuthURL   string `gcfg:"auth-url"`
+		Region    string `gcfg:"region"`
+		AccessKey string `gcfg:"access-key"`
+		SecretKey string `gcfg:"secret-key"`
+		ProjectID string `gcfg:"project-id"`
```
[the remark this answers is not in the document]
Comment (chengxiangdong) [reply pkg/config/cloud.go:26]: it should be useful, i will add this later.

File: pkg/config/cloud.go [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+	"os"
+
+	"github.com/chnsz/golangsdk"
+	"github.com/chnsz/golangsdk/openstack"
+
+	"github.com/huaweicloud/huaweicloud-csi-driver/pkg/utils"
+)
+
+const (
+	UserAgent = "kubernetes-csi"
```
[the remark this answers is not in the document]
Comment (chengxiangdong) [reply pkg/config/cloud.go:15]: ok~

**Pull request merged**